gef.py

```diff
@@ -3555,60 +3435,60 @@ def get_terminal_size():
             return 600, 100
 
 
-def get_generic_arch(module, prefix, arch, mode, big_endian, to_string=False):
+def get_generic_arch(module: ModuleType, prefix: str, arch: str, mode: Optional[str], big_endian: Optional[bool], to_string: bool = False) -> Tuple[str, Union[int, str]]:
     """
     Retrieves architecture and mode from the arguments for use for the holy
     {cap,key}stone/unicorn trinity.
     """
     if to_string:
-        arch = "{:s}.{:s}_ARCH_{:s}".format(module.__name__, prefix, arch)
+        arch = f"{module.__name__}.{prefix}_ARCH_{arch}"
         if mode:
-            mode = "{:s}.{:s}_MODE_{:s}".format(module.__name__, prefix, str(mode))
+            mode = f"{module.__name__}.{prefix}_MODE_{mode}"
         else:
             mode = ""
         if is_big_endian():
-            mode += " + {:s}.{:s}_MODE_BIG_ENDIAN".format(module.__name__, prefix)
+            mode += f" + {module.__name__}.{prefix}_MODE_BIG_ENDIAN"
         else:
-            mode += " + {:s}.{:s}_MODE_LITTLE_ENDIAN".format(module.__name__, prefix)
+            mode += f" + {module.__name__}.{prefix}_MODE_LITTLE_ENDIAN"
 
     else:
-        arch = getattr(module, "{:s}_ARCH_{:s}".format(prefix, arch))
+        arch = getattr(module, f"{prefix}_ARCH_{arch}")
         if mode:
-            mode = getattr(module, "{:s}_MODE_{:s}".format(prefix, mode))
+            mode = getattr(module, f"{prefix}_MODE_{mode}")
         else:
             mode = 0
         if big_endian:
-            mode |= getattr(module, "{:s}_MODE_BIG_ENDIAN".format(prefix))
+            mode |= getattr(module, f"{prefix}_MODE_BIG_ENDIAN")
         else:
-            mode |= getattr(module, "{:s}_MODE_LITTLE_ENDIAN".format(prefix))
+            mode |= getattr(module, f"{prefix}_MODE_LITTLE_ENDIAN")
 
     return arch, mode
 
 
-def get_generic_running_arch(module, prefix, to_string=False):
+def get_generic_running_arch(module: ModuleType, prefix: str, to_string: bool = False) -> Union[Tuple[None, None], Tuple[str, Union[int, str]]]:
     """
     Retrieves architecture and mode from the current context.
     """
 
     if not is_alive():
         return None, None
 
-    if current_arch is not None:
-        arch, mode = current_arch.arch, current_arch.mode
+    if gef.arch is not None:
+        arch, mode = gef.arch.arch, gef.arch.mode
     else:
         raise OSError("Emulation not supported for your OS")
 
     return get_generic_arch(module, prefix, arch, mode, is_big_endian(), to_string)
```

Comment:
deprecated: `is_big_endian()`

```suggestion
    is_big_endian =  gef.arch.endianness == Endianness.BIG_ENDIAN:
    return get_generic_arch(module, prefix, arch, mode, is_big_endian, to_string)
```